src/Core/Cache/Type/RedisCache.php

```diff
@@ -57,22 +57,27 @@ public function __construct(string $hostname, IManageConfigValues $config)
 		$redis_host = $config->get('system', 'redis_host');
 		$redis_port = $config->get('system', 'redis_port');
 		$redis_pw   = $config->get('system', 'redis_password');
-		$redis_db   = $config->get('system', 'redis_db', 0);
+		$redis_db   = (int)$config->get('system', 'redis_db', 0);
 
 		try {
-
-			if (!empty($redis_port) && !@$this->redis->connect($redis_host, $redis_port)) {
-				throw new CachePersistenceException('Expected Redis server at ' . $redis_host . ':' . $redis_port . ' isn\'t available');
-			} else if (!@$this->redis->connect($redis_host)) {
-				throw new CachePersistenceException('Expected Redis server at ' . $redis_host . ' isn\'t available');
+			if (is_numeric($redis_port) && $redis_port > -1) {
```

Comment:
So this will include `0` as well, wouldn't it?

Can you add a test to phpunit as well?

Reply:
I'm sorry, I looked at the `src/Core/Cache/RedisCacheTest.php` file and I'm not sure what I should add or change to test the port = 0 case.

Reply:
I'll try to add it when I'm back from holiday ;-))
--> merge it meanwhile